vm-virtio/src/vhost_user/handler.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 AND BSD-3-Clause
+
+use super::super::{DescriptorChain, Queue, VirtioInterruptType};
+use super::Error;
+use epoll;
+use vmm_sys_util::eventfd::EventFd;
+
+use super::net::CtlVirtqueue;
+use super::{VIRTIO_NET_CTRL_MQ_VQ_PAIRS_MAX, VIRTIO_NET_CTRL_MQ_VQ_PAIRS_MIN};
+use crate::VirtioInterrupt;
+use std::io;
+use std::os::unix::io::AsRawFd;
+use std::result;
+use std::sync::{Arc, RwLock};
+use vhost_rs::vhost_user::{MasterReqHandler, VhostUserMasterReqHandler};
+use vm_memory::{Bytes, GuestMemoryMmap};
+
+const VIRTIO_NET_CTRL_MQ: u8 = 4;
+const QUEUE_SIZE: usize = 256;
+
+pub type Result<T> = result::Result<T, Error>;
+
+pub struct VhostUserEpollConfig {
+    pub interrupt_cb: Arc<VirtioInterrupt>,
+    pub kill_evt: EventFd,
+    pub vu_interrupt_list: Vec<(EventFd, Queue)>,
+}
+
+pub struct VhostUserEpollHandler {
+    pub vu_epoll_cfg: VhostUserEpollConfig,
+    pub cvq: Option<CtlVirtqueue>,
+    pub mem: Option<Arc<RwLock<GuestMemoryMmap>>>,
+    pub slave_req_handler: Option<MasterReqHandler<dyn VhostUserMasterReqHandler>>,
+}
+
+impl VhostUserEpollHandler {
+    /// Construct a new, empty event handler for vhost-based devices.
+    ///
+    /// # Arguments
+    /// * `vhost_user_interrupt` - EventFd to notify queue event
+    /// * `interrupt_status` - semaphore before triggering interrupt event
+    /// * `interrupt_cdb` EventFd for signaling an interrupt that the guest
+    ///                   driver is listening to
+    /// * `kill_evt` - EventFd used to kill the vhost-user-net device
+    /// * `queues` - queues as sharing memory between master and slave
```

Comment:
None of the documented arguments match the actual code, please fix it.

Reply:
Ditto!